src/test/java/org/apache/cassandra/sidecar/routes/VertxRoutingTest.java

```diff
@@ -0,0 +1,213 @@
+
+    @Test
+    void testHandlersWithSameRouteSameOrder(VertxTestContext context) throws Exception
+    {
+        /*
+         * For handlers added to the routes (but same path) with the same order, it is evaluated as the adding order.
+         */
+        CountDownLatch serverReady = new CountDownLatch(1);
+        Router router = Router.router(vertx);
+        router.get("/endpoint").order(1).handler(ctx -> {
+            ctx.response()
+               .setChunked(true) // required to be `true` for adding data from multiple handlers
+               .write("handler 1\n");
+            ctx.next();
+        });
+        router.get("/endpoint").order(1).handler(ctx -> {
+            ctx.response().end("handler 2");
+        });
+
+        server.requestHandler(router).listen(0, result -> serverReady.countDown());
+        assertThat(serverReady.await(10, TimeUnit.SECONDS))
+        .isTrue()
+        .describedAs("Server should be up");
+
+        asyncVerifyRequest("/endpoint", context, response -> {
+            assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.OK.code());
+            assertThat(response.bodyAsString()).isEqualTo("handler 1\n" +
+                                                          "handler 2");
+        });
+    }
+
+    @Test
+    void testLeveledRoutesWithSameOrder(VertxTestContext context) throws Exception
+    {
+        /*
+         * For the leveled routes that is declared with the same order,
+         * the effective evaluation order is the adding order
+         */
+        CountDownLatch serverReady = new CountDownLatch(1);
+        Router router = Router.router(vertx);
+        router.route().order(1).handler(ctx -> {
+            ctx.response()
+               .setChunked(true) // required to be `true` for adding data from multiple handlers
+               .write("root\n");
+            ctx.next();
+        });
+        router.get("/endpoint").order(1).handler(ctx -> {
+            ctx.response().end("endpoint");
+        });
+
+        server.requestHandler(router).listen(0, result -> serverReady.countDown());
+        assertThat(serverReady.await(10, TimeUnit.SECONDS))
+        .isTrue()
+        .describedAs("Server should be up");
+
+        asyncVerifyRequest("/endpoint", context, response -> {
+            assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.OK.code());
+            assertThat(response.bodyAsString()).isEqualTo("root\n" +
+                                                          "endpoint");
+        });
+    }
+
+    @Test
+    void testLeveledRoutesWithSameOrderReversedDeclaration(VertxTestContext context) throws Exception
+    {
+        /*
+         * Similar to testLeveledRoutesWithSameOrder, but the root route is declared after `/endpoint`.
+         * Since `/endpoint` ends the response and does not forward the evaluation,
+         * the root route handler is not called.
+         */
+        AtomicBoolean rootEvaluated = new AtomicBoolean(false);
```

Comment:
alternatively checkpoints can be used https://vertx.io/docs/vertx-junit5/java/#_checkpoint_when_there_are_multiple_success_conditions. No changes needed